### PR TITLE
Fix block enqueue gap and misleading name

### LINKED ISSUE #1136: child_block_minimal_enqueue_gap config is misinterpreted by the BlockQueue
https://github.com/omisego/elixir-omg/blob/master/apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex#L366

should be `>=` not `>`. `child_block_minimal_enqueue_gap` is the minimum that we want "now" and "last submitted" to be apart. So if it's `1`, "now" is `15`, "last submitted" is `14`, we should form and submit a new block.

This causes the current prevalent configuration entry (`1`) to cause the child chain to form & submit every second ethereum block.

Maybe, some reworking of the respective unit tests (that put a stamp on this wrong behavior) is advised, so that the correct behavior is more apparent.

### Bonus

Let's change the confusing config name. I vote for Sirn's idea: `block_submit_every_nth`. `block_submit_every_nth==1` means every ethereum block.

### PR DESCRIPTION
**Pull request by pdobacz**
fixes #1136

## Overview

see issue. This PR limits itself to just fixing the issue at hand. A follow-up PR is prepared with some more far-reaching refactors in the `BlockQueue` to compliment this.

## Changes

- rename `child_block_minimal_enqueue_gap` to `block_submit_every_nth` and make it mean that we want block-every-block if is equal 1
- quickly amended tests to pass and reflect the desired behavior (more cleanup in the follow-up pr)
- _unfortunately_ amending tests required 2 cleanups to be done - 1/ un-fixture the test and 2/ un-import, hope this isn't too much of a distraction

## Testing

End result should be tested via setting up a local child chain and sending a stream of txs to it. It should not exhibit the "every 2nd block" behavior it used to.

But, actually I tried setting up a local env to run this, but:
 - `manual_service_startup.md` is gone and no longer supported
 - running via docker-compose gives me a setup which obfuscates this - the blocks on `geth` are mined every second but the `block_queue_eth_height_check_interval_ms` is set to 6 seconds so there is a ~6 block delay anyway, because eth height isn't checked in between. Not sure how to work around this :thinking: 

EDIT: solved, see comment

### CODE REVIEW

File: apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex [hunk 1]
```diff
@@ -358,20 +358,22 @@ defmodule OMG.ChildChain.BlockQueue.Core do
...
            last_enqueued_block_at_height: last_enqueued_block_at_height,
-           minimal_enqueue_block_gap: minimal_enqueue_block_gap,
+           block_submit_every_nth: block_submit_every_nth,
            wait_for_enqueue: wait_for_enqueue
          },
          is_empty_block
        ) do
-    it_is_time = parent_height - last_enqueued_block_at_height > minimal_enqueue_block_gap
+    # e.g. if we're at 15th Ethereum block now, last enqueued was at 14th, we're submitting a child chain block on every
+    # single Ethereum block (`block_submit_every_nth` == 1), then we should mine a block (`it_is_time` is `true`)
```
Comment (paulperegud) [inline apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex]: To clarify the message, how about using just `enqueue` or just `mine` verb to describe action performed on blocks #14 and #15?

File: apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex [hunk 2]
```diff
@@ -116,7 +116,7 @@ defmodule OMG.ChildChain.BlockQueue.Core do
           # Ethereum height at which first block was mined
           chain_start_parent_height: pos_integer(),
           # minimal gap between child blocks
-          minimal_enqueue_block_gap: pos_integer(),
+          block_submit_every_nth: pos_integer(),
```
Comment (paulperegud) [inline apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex]: Old implementation was inconsistent with pos_integer() type. SMH

File: apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs [hunk 3]
```diff
@@ -186,6 +186,7 @@ defmodule OMG.ChildChain.BlockQueue.CoreTest do
                Core.new(
                  mined_child_block_num: 2000,
                  known_hashes: [{1000, <<1::size(256)>>}],
+                 # TODO(pdobacz, fixing in a follow-up PR): shouldn't be 1 below?
```
Comment (paulperegud) [inline apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs]: No, I think it is consistent with `mined_child_block_num: 2000`. This scenario can be interpreted as "block was mined and submitted, but write to DB has failed" or alternatively "someone else has submitted the block for us".

File: apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex (hunk 2 above)
Comment (pdobacz) [reply apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex]: whoops, I noticed the comment above is stale lol

File: apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex (hunk 1 above)
Comment (pdobacz) [reply apps/omg_child_chain/lib/omg_child_chain/block_queue/core.ex]: these mean a different thing. But, current wording is indeed wrong, should be `then we should form a block`, mining is confusing.

But then `enqueue` vs `form` is significant. We `form` if a certain amount of eth blocks are `mined` after our last child block `enqueued`.

File: apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs (hunk 3 above)
Comment (pdobacz) [reply apps/omg_child_chain/test/omg_child_chain/block_queue/core_test.exs]: ok gotcha, you're right. I'll then keep both this version and one with `1` in the hash. I think both tests are relevant. And I'll ditch the duplicate https://github.com/omisego/elixir-omg/pull/1253/files#diff-517a845b00e424eeec3da45ca99c9450R218

**Comment by pdobacz** [comment]
I managed to run the services locally and test. It does indeed submit a block every eth block:

```
childchain_1    | 2020-01-13 17:15:37.084 [info] module=OMG.ChildChain.BlockQueue.Core function=log_success/2 ⋅Submitted %OMG.ChildChain.BlockQueue.Core.BlockSubmission{gas_price: 6276211922, hash: <<117, 249, 155, 25, 16, 35, 5, 158, 19, 207, 7, 170, 162, 47, 244, 137, 209, 125, 252, 130, 249, 158, 95, 100, 108, 155, 214, 164, 193, 167, 123, 55>>, nonce: 13, num: 13000} at: <<215, 14, 11, 2, 251, 231, 59, 0, 100, 77, 89, 249, 136, 161, 26, 198, 247, 48, 7, 29, 243, 58, 201, 215, 42, 86, 189, 1, 60, 165, 111, 12>>⋅
geth_1          | INFO [01-13|17:15:38.000] Successfully sealed new block            number=372 sealhash=996b83…7415f1 hash=870bb8…08f8fd elapsed=998.928ms
geth_1          | INFO [01-13|17:15:38.000] 🔗 block reached canonical chain          number=365 hash=083f74…354d10
geth_1          | INFO [01-13|17:15:38.000] 🔨 mined potential block                  number=372 hash=870bb8…08f8fd
geth_1          | INFO [01-13|17:15:38.001] Commit new mining work                   number=373 sealhash=d1efa5…d862e4 uncles=0 txs=0 gas=0      fees=0               elapsed=349.251µs
geth_1          | INFO [01-13|17:15:38.001] Commit new mining work                   number=373 sealhash=339e0d…119a09 uncles=0 txs=1 gas=76319  fees=0.0004789942177 elapsed=786.102µs
childchain_1    | 2020-01-13 17:15:38.081 [info] module=OMG.ChildChain.BlockQueue.Server function=handle_info/2 ⋅Enqueuing block num '14000', hash '"0xb32e66d1e0d25afa1ac801c88ca92d291067a8726f533dcc7c8de52edd10440e"'⋅
geth_1          | INFO [01-13|17:15:38.084] Submitted transaction                    fullhash=0xd02ae0a7fd11b6223edcfd378780661ece9c127eea4debc5ebaf132d50b14d05 recipient=0xC673e4ffcB8464FaFf908A6804FE0e635AF0ea2f
childchain_1    | 2020-01-13 17:15:38.086 [info] module=OMG.ChildChain.BlockQueue.Core function=log_success/2 ⋅Submitted %OMG.ChildChain.BlockQueue.Core.BlockSubmission{gas_price: 5648590730, hash: <<179, 46, 102, 209, 224, 210, 90, 250, 26, 200, 1, 200, 140, 169, 45, 41, 16, 103, 168, 114, 111, 83, 61, 204, 124, 141, 229, 46, 221, 16, 68, 14>>, nonce: 14, num: 14000} at: <<208, 42, 224, 167, 253, 17, 182, 34, 62, 220, 253, 55, 135, 128, 102, 30, 206, 156, 18, 126, 234, 77, 235, 197, 235, 175, 19, 45, 80, 177, 77, 5>>⋅
geth_1          | INFO [01-13|17:15:39.000] Successfully sealed new block            number=373 sealhash=339e0d…119a09 hash=f9e1c4…519088 elapsed=999.327ms
geth_1          | INFO [01-13|17:15:39.000] 🔗 block reached canonical chain          number=366 hash=9bbaba…22f5b4
geth_1          | INFO [01-13|17:15:39.000] 🔨 mined potential block                  number=373 hash=f9e1c4…519088
geth_1          | INFO [01-13|17:15:39.001] Commit new mining work                   number=374 sealhash=c286a1…00624c uncles=0 txs=0 gas=0      fees=0               elapsed=396.124µs
geth_1          | INFO [01-13|17:15:39.001] Commit new mining work                   number=374 sealhash=2c157e…e90d7f uncles=0 txs=1 gas=76319  fees=0.0004310947959 elapsed=839.379µs
childchain_1    | 2020-01-13 17:15:39.077 [info] module=OMG.ChildChain.BlockQueue.Server function=handle_info/2 ⋅Enqueuing block num '15000', hash '"0x4685894f0235287a0cb6382f8d79d378d60828522edc231d30cfd970c91e6794"'⋅
geth_1          | INFO [01-13|17:15:39.080] Submitted transaction                    fullhash=0xde0dfce89b50e4be71f83093f3d20572bb94f88da4022e90eb9319d890bb2f72 recipient=0xC673e4ffcB8464FaFf908A6804FE0e635AF0ea2f
```

(note for future self: one needs to shorten the `block_queue_eth_height_check_interval_ms` setting  to `500` in code and use `make docker-update-child_chain` to build an image off of the modified code, to match the 1-second mining interval of `geth`)

**Pull request merged**